src/main/java/com/limechain/trie/AccessorHolder.java

```diff
@@ -30,7 +32,9 @@ public static AccessorHolder getInstance() {
      * @return         The BlockTrieAccessor instance representing the block trie with the specified root hash.
      */
     public BlockTrieAccessor setToStateRoot(byte[] lastRoot) {
-        this.blockTrieAccessor = new BlockTrieAccessor(lastRoot);
+        TrieStorage trieStorage = AppBean.getBean(TrieStorage.class);
```

Comment:
That's a weird place: a boundary for tapping into spring beans. 

The static `AccessorHolder` singleton relies on the `TrieStorage` from the spring context.

However, the `AccessorHolder` is, now entirely obsolete IMO, since the problem it was created to solve (circumnavigate sprawling singletons) is solved in this PR. However, this class falls under general between-blocks logic, it's out of scope of the current PR.

Reply:
Delegated to the follow-up PR.

Reply:
`AccessorHolder` was entirely removed as not needed. Fixed in [e222d87](https://github.com/LimeChain/Fruzhin/pull/464/commits/e222d870a46afcb797ef1ab9b9517f2b8c8bd6dc)